benchmark_runner/common/grafana/update_grafana_last_value_mappings.py

```diff
@@ -0,0 +1,117 @@
+
+import json
+from datetime import datetime, timedelta
+
+from benchmark_runner.main.environment_variables import environment_variables
+from benchmark_runner.common.elasticsearch.elasticsearch_operations import ElasticSearchOperations
+
+
+class UpdateGrafanaLastValueMappings:
+    """
+    This class updates grafana dashboard with last value mappings from ElasticSearch
+    """
+    def __init__(self, main_libsonnet_path: str):
+        self._environment_variables_dict = environment_variables.environment_variables_dict
+        self.elasticsearch = ElasticSearchOperations(es_host=self._environment_variables_dict.get('elasticsearch', ''),
+                                                     es_port=self._environment_variables_dict.get('elasticsearch_port', ''),
+                                                     es_user=self._environment_variables_dict.get('elasticsearch_user', ''),
+                                                     es_password=self._environment_variables_dict.get('elasticsearch_password', ''))
+        self.main_libsonnet_path = main_libsonnet_path
+        self.value_mappings = self.get_value_mappings()
+
+    def get_last_elasticsearch_versions(self, last_days=7):
+        """
+        This method fetched new version from ElasticSearch
+        :param last_days: default fetch 7 days
+        :return:
+        """
+        new_versions = {}
+        current_datetime = datetime.now()
+        start_datetime = current_datetime - timedelta(days=last_days)
+
+        ids = self.elasticsearch.get_index_ids_between_dates(index='ci-status', start_datetime=start_datetime,
+                                                        end_datetime=current_datetime)
+
+        display_versions = ['ocp_version', 'cnv_version', 'kata_version', 'kata_rpm_version', 'odf_version']
+        for id in ids:
+            data = self.elasticsearch.get_elasticsearch_index_by_id(index='ci-status', id=id)
+            for version, value in data['_source'].items():
+                if version in display_versions and value not in new_versions.values():
+                    new_versions[value.replace(".", "").replace("-", "")] = value
+
+        return new_versions
+
+    def get_value_mappings(self):
+        """
+        This method gets value mappings from main.libsonnet
+        :return:
+        """
+        # Pattern to check in each line
+        start = "        stateTimeline.valueMapping.RegexMap.withOptions("
+        end = "+ stateTimeline.valueMapping.RegexMap.withType('value')"
+
+        # Initialize a flag to indicate whether the lines are within the desired range
+        within_range = False
+        mapping_lines = []
+
+        # Read the content of the file and store lines between start and end patterns in a list
+        with open(self.main_libsonnet_path, 'r') as file:
```

Comment:
Ugh.  You're editing Python code this way?  If that's what you're doing, I suggest using a clearly named comment that won't be duplicated at the end of the line, something like
```
# START_92734987
```
or some other fixed cookie.

Reply:
Good point, fix it by adding
```
        //START_VALUE_MAPPING_227
        //END_VALUE_MAPPING_227

```